### PR TITLE
Add new confval "author" and use it for lazy evaluation

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Feature

### Purpose
- Add new configure value `author`
- Use it for lazy evaluation

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -20,6 +23,7 @@ Features added
 * Add :event:`config-inited` event
 * Add ``sphinx.config.Any`` to represent the config value accepts any type of
   value
+* Add :confval:`author` as a configuration value
```
Comment (tk0miya) [inline CHANGES]: So far, `author` is defined as local variable in `conf.py`.
This enables all of Sphinx modules can refer it.

File: sphinx/builders/epub3.py [hunk 2]
```diff
@@ -231,9 +231,9 @@ def setup(app):
     app.add_config_value('epub_theme', 'epub', 'html')
     app.add_config_value('epub_theme_options', {}, 'html')
     app.add_config_value('epub_title', lambda self: self.html_title, 'html')
-    app.add_config_value('epub_author', 'unknown', 'html')
+    app.add_config_value('epub_author', lambda self: self.author, 'html')
```
Comment (tk0miya) [inline sphinx/builders/epub3.py:234]: This is same as what quickstart does.
https://github.com/sphinx-doc/sphinx/blob/77ae4249c62bebb137b29b1892a6b89b70c984b5/sphinx/templates/quickstart/conf.py_t#L173-L177

So I will be able to omit them from `conf.py`.

**Review by shimizukawa — APPROVED** [review]
LGTM!

**Comment by tk0miya** [comment]
Thanks!

**Pull request merged**